### PR TITLE
Added Microsoft Teams App Rule

### PR DESCRIPTION
**Pull request by Cliveo**
simple navigation, general commands and communication management

# Title

Simple mapping the rule for Microsoft teams app

## Description

Just a bunch of key bindings

## Related Issue

no Issue

## Motivation and Context

becoming quite popular tool for use at work, 

## How Has This Been Tested

tested in application

## Types of changes

1 new mapping rule

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue or bug)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Renamed existing command phrases (we discourage this without a strong rationale).

## Checklist

- [ ] I have read the CONTRIBUTING document.
- [ ] My code follows the code style of this project.
- [x] I have checked that my code does not duplicate functionality elsewhere in Caster.
- [ ] I have checked for and utilized existing command phrases from within Caster (delete if not applicable). 
- [x] My code implements all the features I wish to merge in this pull request.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests pass.

## Maintainer/Reviewer Checklist

- [ ] Basic functionality has been tested and works as claimed.
- [ ] New documentation is clear and complete.
- [ ] Code is clear and readable.

### CODE REVIEW

**Comment by LexiconCode** [comment]
Looks great! I can update the documentation in [Application_Commands_Quick_Reference](https://github.com/dictation-toolbox/Caster/blob/master/docs/readthedocs/Application_Commands_Quick_Reference.md) sometime tomorrow.

File: castervoice/rules/apps/chat/MSTeamsRule.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
+from castervoice.lib.actions import Key, Text, Mouse
+from castervoice.lib.ctrl.mgr.rule_details import RuleDetails
+from castervoice.lib.merge.additions import IntegerRefST
+from castervoice.lib.merge.state.short import R
+from dragonfly import (AppContext, Choice, Dictation, Function, MappingRule,
+                       Repeat)
+
+class MSTeamsRule(MappingRule):
+    name = "microsoft teams"
```
Comment (LexiconCode) [inline castervoice/rules/apps/chat/MSTeamsRule.py]: `name` isn't required here only in RuleDetails for MappingRule's.

Comment (Cliveo) [reply castervoice/rules/apps/chat/MSTeamsRule.py]: Thanks that's good to know, i've got a bunch of other commands now that i'm using, is there a reason to why windows key bindings are pretty light? i'm not sure if others will find them useful but i have definitely have. E.g. 
        "open explorer":
            R(Key("w-e")),
        "show desktop":
            R(Key("w-d")),
        "windows speech":
            R(Key("w-ctrl")),
        "admin tools":
            R(Key("w-x")),
        "windows run":
            R(Key("w-r")),
        "press windows":
            R(Key("win")),
        "windows search":
            R(Key("w-s")),
        "taskbar":
            R(Key("w-t")),
        "taskbar <n>":
            R(Key("w-%(n)s")),
        "taskbar menu <n>":
            R(Key("wa-%(n)s")),

Comment (LexiconCode) [reply castervoice/rules/apps/chat/MSTeamsRule.py]: The Windows keyboard bindings could be added.

**Comment by LexiconCode** [comment]
Turns out I won't be able to get to this tonight but tomorrow sometime.

**Comment by LexiconCode** [comment]
added docs and to unit tests d136856bcbead0a45a25f4eb208dde0f8a35f96f

**Pull request merged**